pkg/auth/oidc/common.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package oidc
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/Peripli/service-manager-cli/internal/util"
+	"github.com/Peripli/service-manager-cli/pkg/auth"
+	"github.com/Peripli/service-manager-cli/pkg/httputil"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// NewClient builds configured HTTP client.
+//
+// If token is provided will execute try to refresh the token if it has expired,
+// if not provided will do client_credentials flow and fetch token
+func NewClient(options *auth.Options, token *auth.Token) auth.Client {
+	httpClient := util.BuildHTTPClient(options.SSLDisabled)
```

Comment:
the logic here seems to semi-duplicate the logic in the strategies. Once again can we avoid the duplication?